org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/ui/tests/quickfix/AssistQuickFixTest.java

```diff
@@ -10436,6 +10436,74 @@ public void testDoesntOfferConvertToStaticImportForImportDeclarations() throws E
 		assertProposalDoesNotExist(proposals, "Convert to static import (replace all occurrences)");
 	}
 
+	@Test
+	public void testConvertToStaticImportFromReferenceToSubclass() throws Exception {
+		IPackageFragment pack1= fSourceFolder.createPackageFragment("test1", false, null);
+		StringBuilder buf= new StringBuilder();
+		buf.append("package test1;\n");
+		buf.append("public class T {\n");
+		buf.append("    public static void foo() { };\n");
+		buf.append("}\n");
+		pack1.createCompilationUnit("T.java", buf.toString(), false, null);
+
+		buf= new StringBuilder();
+		buf.append("package test1;\n");
+		buf.append("public class TSub extends T {\n");
+		buf.append("}\n");
+		pack1.createCompilationUnit("TSub.java", buf.toString(), false, null);
+
+
+		IPackageFragment pack2= fSourceFolder.createPackageFragment("test2", false, null);
+		buf= new StringBuilder();
+		buf.append("package test2;\n");
+		buf.append("\n");
+		buf.append("import test1.TSub;\n");
+		buf.append("public class S {\n");
+		buf.append("    public S() {\n");
+		buf.append("        TSub.foo();\n");
+		buf.append("        TSub.foo();\n");
+		buf.append("    }\n");
+		buf.append("}\n");
+		ICompilationUnit cu= pack2.createCompilationUnit("S.java", buf.toString(), false, null);
+
+		String selection= "foo";
+		int offset= buf.toString().indexOf(selection);
+		AssistContext context= getCorrectionContext(cu, offset, selection.length());
+		ArrayList<IJavaCompletionProposal> proposals= collectAssists(context, false);
+
+		assertProposalExists(proposals, "Convert to static import");
+		assertProposalExists(proposals, "Convert to static import (replace all occurrences)");
```

Comment:
Hello, the proposal name must be the NLS translatable string.  Same for your calls to assertProposalPreviewEquals below.  See other examples of assertProposalExists and assertProposalPreviewEquals in this file.  If you have issues doing this, let me know and I can assist.

Reply:
Oh sorry, I just copied from my earlier tests for that quick fix. Should be fixed?

Reply:
No problem.  The earlier tests should probably be fixed in a future patch.  I changed your patch slightly to remove the requirement of the ordering of the proposals (not essential, but makes the test more robust).  Will merge as soon as tests run.